src/LibHdb++Cassandra.cpp

```diff
@@ -1119,637 +1286,489 @@ void HdbPPCassandra::extract_and_bind_rw_values(CassStatement* statement,
+	cass_statement_bind_string(statement,13, data->attr_conf->description.c_str()); // description
+	cass_statement_set_consistency(statement, CASS_CONSISTENCY_LOCAL_QUORUM); // TODO: Make the consistency tunable? => would be useful!
+
+	CassError rc = execute_statement(statement);
+
+	if (rc != CASS_OK)
+		throw_execute_exception("ERROR executing insert query", query_str.str(), rc, __func__);
+
+	TRACE_MSG("Leaving");
+}
 
-int HdbPPCassandra::configure_Attr(string name, 
-                                   int type/*DEV_DOUBLE, DEV_STRING, ..*/,
-                                   int format/*SCALAR, SPECTRUM, ..*/,
-                                   int write_type/*READ, READ_WRITE, ..*/,
-                                   unsigned int ttl /* hours, 0=infinity*/)
+void HdbPPCassandra::configure_Attr(string name,
+								   int type/*DEV_DOUBLE, DEV_STRING, ..*/,
+								   int format/*SCALAR, SPECTRUM, ..*/,
+								   int write_type/*READ, READ_WRITE, ..*/,
+								   unsigned int ttl /* hours, 0=infinity*/)
 {
+	TRACE_MSG("Entering");
+
 	string facility = get_only_tango_host(name);
 	facility = add_domain(facility);
 	string attr_name = get_only_attr_name(name);
 	string domain = "";
 	string family = "";
 	string member = "";
 	string attribute_name = "";
-	int parsingError = getDomainFamMembName(attr_name,domain,family,member,attribute_name);
-	if(parsingError)
+
+	if(getDomainFamMembName(attr_name,domain,family,member,attribute_name) < 0)
 	{
-		cerr << "Error parsing attribute name \"" << attr_name << "\"" << endl;
-		return -5;
+		stringstream error_desc;
+		error_desc << "ERROR parsing attribute name  \"" << attr_name << "\": " << ends;
+		ERROR_MSG(error_desc.str() << endl);
+		Tango::Except::throw_exception("Parse Error", error_desc.str().c_str(), __func__);
 	}
-#ifdef _CASS_LIB_DEBUG
-	cout<< __func__ << ": name="<<name<<" -> facility="<<facility<<" attr_name="<<attr_name<< endl;
-#endif
+
+	DEBUG_MSG("name="<<name<<" -> facility="<<facility<<" attr_name="<<attr_name<< endl);
+
 	string data_type = get_data_type(type, format, write_type);
 	CassUuid id;
 	unsigned int conf_ttl = 0;
-	int ret = find_attr_id_type_and_ttl(facility, attr_name, id, data_type, conf_ttl);
+	FindAttrResult find_attr_result = find_attr_id_type_and_ttl(facility, attr_name, id, data_type, conf_ttl);
 	//ID already present but different configuration (attribute type)
-	if(ret == -2)
+	if(find_attr_result == FoundAttrWithDifferentType)
 	{
-		cerr << __func__ << ": "<< facility << "/" << attr_name << " already configured with different configuration" << endl;
-		cerr << "Please contact your administrator for this special case" << endl;
+		ERROR_MSG(facility << "/" << attr_name << " already configured with different configuration" << endl);
+		ERROR_MSG("Please contact your administrator for this special case" << endl);
 		// Need to contact an administrator in this case!?
+		// TODO (SJ) How is this handled?
```

Comment:
Now that we have support for exceptions, I think an exception should be thrown here.